MAVProxy/modules/mavproxy_battery.py

```diff
@@ -69,14 +69,26 @@ def battery_report(self):
             if batt_mon == 0:
                 continue
 
-            #report voltage level only
-            if batt_mon == 3:
-                battery_string += 'Batt%u: %.2fV ' % (id+1, self.voltage_level[id])
-            elif batt_mon >= 4:
-                battery_string += 'Batt%u: %u%%/%.2fV %.1fA ' % (id+1,
-                                                                   self.battery_level[id],
-                                                                   self.voltage_level[id],
-                                                                   self.current_battery[id])
+            #show full battery voltage when number of cells is not set
+            if self.numcells(id) == 0:
```

Comment:
This patch can be simplified by continuing the `continue` pattern that's already in here.  So you would add a new block looking something like this, about line 72:

```
          # show per-cell voltage if number of cells is set:
          if self.numcells(id) != 0:
               #this is a workaround for the race condition since self.per_cell only gets filled in once the setting is set
                if not id in self.per_cell:
                    continue
                if batt_mon == 3:
                    battery_string += 'Batt%u: %.2fV*%ucells ' % (id+1, self.per_cell[id], self.numcells(id))
                    continue
                if batt_mon >= 4:
                    battery_string += 'Batt%u: %u%%/%.2fV*%ucells %.1fA ' % (id+1, self.battery_level[id], self.per_cell[id], self.numcells(id), self.current_battery[id])
                continue
```

The chief advantage of this is that your patch is simply added lines, not changing existing lines.

Reply:
Done.